purs/src/Component/BMark.purs

```diff
@@ -1,262 +1,262 @@
+        , br_
+        , div [ class_ "description mt1 mid-gray" ] [ HH.slot _markdown unit Markdown.component bm.description absurd ]
+        , div [ class_ "tags" ] $
+              whenA (not (S.null bm.tags)) $ \_ ->
+                map (\tag -> a [ class_ ("link tag mr1" <> guard (S.take 1 tag == ".") " private")
+                             , href (linkToFilterTag tag) ]
+                             [ text tag ])
+                (S.split (Pattern " ") bm.tags)
+
+        , a [ class_ "link f7 dib gray w4", href (linkToFilterSingle bm.slug), title shdatetime ]
+          [ text shdate ]
+
+        -- links
+        , whenH app.dat.isowner $ \_ ->
+            div [ class_ "edit_links di" ]
+            [ button [ type_ ButtonButton, onClick \_ -> Just (BEdit true), class_ "edit light-silver hover-blue" ] [ text "edit  " ]
+            , div [ class_ "delete_link di" ]
+              [ button [ type_ ButtonButton, onClick \_ -> Just (BDeleteAsk true), class_ ("delete light-silver hover-blue" <> guard s.deleteAsk " dn") ] [ text "delete" ]
+              , span ([ class_ ("confirm red" <> guard (not s.deleteAsk) " dn") ] )
+                [ button [ type_ ButtonButton, onClick \_ -> Just (BDeleteAsk false)] [ text "cancel / " ]
+                , button [ type_ ButtonButton, onClick \_ -> Just BDestroy, class_ "red" ] [ text "destroy" ]
+                ]
+              ]
+            ]
+        , whenH app.dat.isowner $ \_ ->
+            div [ class_ "read di" ] $
+              guard bm.toread
+              [ text "  "
+              , button [ onClick \_ -> Just BMarkRead, class_ "mark_read" ] [ text "mark as read"]
+              ]
+        ]
+
+
+     display_edit _ =
+       div [ class_ "edit_bookmark_form pa2 pt0 bg-white" ] $
+       [ form [ onSubmit (Just <<< BEditSubmit) ]
+         [ div_ [ text "url" ]
+         , input [ type_ InputUrl , class_ "url w-100 mb2 pt1 edit_form_input" , required true , name "url"
+                 , value (edit_bm.url) , onValueChange (editField Eurl) ]
+         , div_ [ text "title" ]
+         , div [class_ "flex"]
+               [input [ type_ InputText , class_ "title w-100 mb2 pt1 edit_form_input" , name "title"
+                      , value (edit_bm.title) , onValueChange (editField Etitle) ]
+               , button [ disabled s.loading, type_ ButtonButton, onClick \_ -> Just BLookupTitle, class_ ("ml1 pa1 mb2 dark-gray ba b--moon-gray bg-near-white pointer rdim f7 " <> guard s.loading "bg-light-silver") ] [ text "fetch" ]
+               ]
+         , div_ [ text "description" ]
+         , textarea [ class_ "description w-100 mb1 pt1 edit_form_input" , name "description", rows 5
+                    , value (edit_bm.description) , onValueChange (editField Edescription) ]
+         , div [ id_ "tags_input_box"]
+           [ div_ [ text "tags" ]
+             , input [ id_ (tagid edit_bm), type_ InputText , class_ "tags w-100 mb1 pt1 edit_form_input" , name "tags"
```

Comment:
I believe this diff is so difficult to read because of the whitespace trimming that my formatter performed. In these purscript files I've only removed the `f7` class from the inputs and textareas.

Reply:
The changes in these files require additional work in order to reflect the build. I wasn't sure how the frontend build works - do you run the `parcel-build` script and then copy `dist` over to `static/js`?

Reply:
in the `purs/` folder I just run `make`, which should automatically copy the build output into `../static/js`

Reply:
Oh I see it now, thank you